File: Crio/crio_programming_interview_problems-master/ValidAnagram/ValidAnagram.cpp
#include <bits/stdc++.h>
#include "FastIO.hpp"

using namespace std;

class ValidAnagram {
    public:
        bool isAnagram(string s, string t) {
            // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
            map<char,int> mp,mp2;
            for(auto i:s) mp[i]++;
            for(auto i:t) mp2[i]++;
            for(auto i:mp)
            {
                if(mp[i.first] != mp2[i.first]) return false;
            }
            for(auto i:mp2)
            {
                if(mp[i.first] != mp2[i.first]) return false;
            }
            return true;
            // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
        }
};

int main() {
    FastIO();
    string s, t;
    getline(cin, s);
    getline(cin, t);
    bool result = ValidAnagram().isAnagram(s, t);
    cout<< boolalpha << result;
    return 0;
}

File: Crio/crio_programming_interview_problems-master/ValidAnagram/ValidAnagram.py
def validAnagram(s, t):
    # CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    if(len(s)!=len(t)):
        return False
    m = [0]*26
    for ch in s:
        m[ord(ch)-ord('a')]+=1
    for ch in t:
        if(m[ord(ch)-ord('a')]>0):
            m[ord(ch)-ord('a')]-=1
        else:
            return False
    return True
    # CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

if __name__ == '__main__':
    s = input()
    t = input()
    if validAnagram(s, t):
        print('true')
    else:
        print('false')
